src/main/java/com/fasterxml/jackson/core/base/ParserBase.java

```diff
@@ -819,6 +824,80 @@ public BigDecimal getDecimalValue() throws IOException
         return _getBigDecimal();
     }
 
+    @Override
+    public LazyNumber getLazyNumber() throws IOException
+    {
+        LazyNumber lazyNumber = null;
+        if ((_numTypesValid & NR_BIGDECIMAL) == 0) {
+            if (_numTypesValid == NR_UNKNOWN) {
+                _parseNumericValue(NR_BIGDECIMAL);
+            }
+            if (_numberString != null) {
+                lazyNumber = new LazyBigDecimal(_numberString,
+                        isEnabled(StreamReadFeature.USE_FAST_BIG_NUMBER_PARSER));
+                _numTypesValid |= NR_BIGDECIMAL;
+            } else {
+                lazyNumber = lazyNumberFromParsedPrimitiveNumber();
+            }
+        } else if ((_numTypesValid & NR_DOUBLE) == 0) {
+            if (_numTypesValid == NR_UNKNOWN) {
+                _parseNumericValue(NR_DOUBLE);
+            }
+            if (_numberString != null) {
+                lazyNumber = new LazyDouble(_numberString,
+                        isEnabled(StreamReadFeature.USE_FAST_DOUBLE_PARSER));
+                _numTypesValid |= NR_DOUBLE;
+            } else {
+                lazyNumber = lazyNumberFromParsedPrimitiveNumber();
+            }
+        } else if ((_numTypesValid & NR_FLOAT) == 0) {
+            if (_numTypesValid == NR_UNKNOWN) {
+                _parseNumericValue(NR_FLOAT);
+            }
+            if (_numberString != null) {
+                lazyNumber = new LazyFloat(_numberString,
+                        isEnabled(StreamReadFeature.USE_FAST_DOUBLE_PARSER));
+                _numTypesValid |= NR_FLOAT;
+            } else {
+                lazyNumber = lazyNumberFromParsedPrimitiveNumber();
+            }
+        } else if ((_numTypesValid & NR_BIGINT) == 0) {
+            if (_numTypesValid == NR_UNKNOWN) {
+                _parseNumericValue(NR_BIGINT);
+            }
+            if (_numberString != null) {
+                lazyNumber = new LazyBigInteger(_numberString,
+                        isEnabled(StreamReadFeature.USE_FAST_BIG_NUMBER_PARSER));
+                _numTypesValid |= NR_BIGINT;
+            } else {
+                lazyNumber = lazyNumberFromParsedPrimitiveNumber();
+            }
+        } else {
+            throw new JsonParseException(this, "unable to convert value to LazyNumber: " + getText());
+        }
+        return lazyNumber;
+    }
+
+    private LazyNumber lazyNumberFromParsedPrimitiveNumber() throws IOException {
+        switch (_numTypesValid) {
```

Comment:
Cannot use `switch` as `_numTypesValid` is a bitset, need to check with `&`.
(unfortunately it would probably work in most tested cases as access using multiple getXxx() methods is rare).

Reply:
removed this method